webextension/background.js

```diff
@@ -289,6 +289,44 @@ const backgroundLogic = {
     });
   },
 
+  openTab(options) {
+    const url = options.url || undefined;
+    const userContextId = ("userContextId" in options) ? options.userContextId : 0;
+    const active = ("nofocus" in options) ? options.nofocus : true;
+/*
+TODO check if we are opening in the correct window
+    return new Promise(resolve => {
+      if ("window" in args && args.window) {
+        resolve(args.window);
+      } else {
+        this._recentBrowserWindow().then(browserWin => {
+          resolve(browserWin);
+        }).catch(() => {});
+      }
+    }).then(browserWin => {
+*/
+    const source = ("source" in options) ? options.source : null;
+
+    // Only send telemetry for tabs opened by UI - i.e., not via showTabs
+    if (source && userContextId) {
+      this.sendTelemetryPayload({
+        "event": "open-tab",
+        "eventSource": source,
+        "userContextId": userContextId,
+//IORE FOR NOW          "clickedContainerTabCount": this._containerTabCount(userContextId)
+      });
+    }
+    // Autofocus url bar will happen in 54: https://bugzilla.mozilla.org/show_bug.cgi?id=1295072
+
+    // What is this doing? promise = this.showTabs(args);
```

Comment:
I couldn't really work out what this was supposed to be doing, however there is telemetry attached to this method when it was index.js